### ISSUE TITLE
ClassCastException with custom MP Rest Client ResponseExceptionMapper implementation

### ISSUE THREAD

**Issue by lreimer** [payara/Payara#3927]
# Description #
----------

We have implemented a custom MicroProfile REST Client `ResponseExceptionMapper` that converts HTTP response codes to custom, checked exceptions.

When the exception mapper is triggered, we see a `ClassCastException` in the (patched) Jersey Client class `org.glassfish.jersey.client.ResponseExceptionMappingStages`. The root cause is the _(wrong?)_ cast to `WebApplicationException` in lines 92..94:

```java
if (prioritised.isPresent()) {
    throw (WebApplicationException) prioritised.get();
}
```

This forces us to make our checked exception a subclass of `WebApplicationException` and thus a runtime exception. Also, it is kind of a mismatch since according to the generic interface definition any `Throwable` is supported for `ResponseExceptionMapper`. 

## Expected Outcome

The desired exception from the custom `ResponseExceptionMapper` is thrown instead of the unexpected `ClassCastException`.

## Current Outcome

A `ClassCastException` in the Jersey Client class `org.glassfish.jersey.client.ResponseExceptionMappingStages` which results in unexpected behaviour when using a MicroProfile REST Client implementation with custom exceptions.

## Steps to reproduce (Only for bug reports) 

Implement a custom `ResponseExceptionMapper` that maps a response to an exception that is not a subclass of `WebApplicationException`.

## Environment ##

- **Payara Version**: 5.191
- **Edition**: Micro
- **JDK Version**: Zulu JDK 8u202
- **Operating System**: Linux

**Comment by smillidge** [comment]
Can you create a simple test case that demonstrates the issue?

**Comment by lreimer** [comment]
Of course, I will be prepare a small test case.

**Comment by pdudits** [comment]
I believe this is fixed in 5.192, which contains entirely different implementation of REST client. Can you try with the new version, please?

**Comment by lreimer** [comment]
Thanks for the info. I will try the latest 5.192 version and report here on the results.
